### PR TITLE
ADD: Courses page + Institutions minor tweaks

### PR DESCRIPTION
**Pull request by XRegiGigaSX**
Made necessary updates: Reduced box shadow, Boxes made rounder, horizontal scroll added.

### CODE REVIEW

**Comment by codadept** [comment]
Take a pull from `dev` branch and push the merge commit

File: src/App.jsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { Route, Routes } from "react-router-dom";
 
 import { Footer, Navbar } from "./Components";
 
+import Courses from "./Components/Courses/Courses";
+
```
Comment (codadept) [inline src/App.jsx]: It should be in Pages!

File: src/Components/Courses/Courses.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+              <img src={bookmark} alt="bookmark" className="courses-card-bm" />
+              <div className="courses-card-subject">{data?.subjectName}</div>
+              <img src={book} alt="book" className="courses-card-img" />
+            </div>
+          );
+        })}
+      </div>
+    </div>
+  );
+};
```
Comment (codadept) [inline src/Components/Courses/Courses.jsx]: You haven't used module.scss!

File: src/Components/Institutions/Institutions.jsx [hunk 3]
```diff
@@ -19,6 +19,10 @@ const dummyData = [
     insti: "Indian Institute of Technology, Bombay",
     pfp: dp,
   },
+  {
+    insti: "Indian Institute of Technology, Bombay",
+    pfp: dp,
+  },
 ];
```
Comment (codadept) [inline src/Components/Institutions/Institutions.jsx]: Use all the data from a json file in public folder

for example if there is a json file, in the path
/public/db/institute.json

Use fetch API to fetch the data in the Page 

```js 
fetch("/db/institute.json")
```

File: src/Components/Courses/Courses.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    {
+      insti: "Indian Institute of Technology, Bombay",
+      subjectName: "Maths",
+    },
+    {
+      insti: "Indian Institute of Technology, Bombay",
+      subjectName: "Physics",
+    },
+  ];
```
Comment (codadept) [inline src/Components/Courses/Courses.jsx]: Use json file in public and fetch API to mimic backend API call for your ease later on after backend integration

**Review by codadept — CHANGES_REQUESTED** [review]
Move Courses and Institutions to Pages! As it is a page for your frontend application

File: src/Components/Courses/Courses.jsx (hunk 4 above)
Comment (XRegiGigaSX) [reply src/Components/Courses/Courses.jsx]: Made this change in the latest commit.

File: src/Components/Institutions/Institutions.jsx (hunk 3 above)
Comment (XRegiGigaSX) [reply src/Components/Institutions/Institutions.jsx]: Implemented fetch API.

File: src/Components/Courses/Courses.jsx (hunk 2 above)
Comment (XRegiGigaSX) [reply src/Components/Courses/Courses.jsx]: Now using module.scss.

File: src/App.jsx (hunk 1 above)
Comment (XRegiGigaSX) [reply src/App.jsx]: Added courses in pages. Institutions was a component, hence left it as it is.

File: src/Pages/Courses/Courses.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { Link } from "react-router-dom";
+import styles from "./Courses.module.scss";
+// import book from "/images/book.png";
+// import bookmark from "/images/bookmark.png";
+// import bookmarkA from "/images/bmActive.png";
+// import backArrow from "/images/arrow.png";
+
+const Courses = () => {
+  const [coursesData, setCoursesData] = useState([]);
+  const [BM, setBM] = useState(false);
```
Comment (codadept) [inline src/Pages/Courses/Courses.jsx]: using only one state to toggle bookmark feature of all courses?

If you bookmark one course, only that course should get bookmarked not all the courses.

Fix this.

**Review by codadept — CHANGES_REQUESTED** [review]

**Comment by codadept** [comment]
Also add navigation links to the courses page as it was supposed to. Check the doc in the WA description on the navigations to different pages.

File: src/Pages/Courses/Courses.jsx (hunk 5 above)
Comment (XRegiGigaSX) [reply src/Pages/Courses/Courses.jsx]: Fixed.

File: src/App.jsx [hunk 6]
```diff
@@ -2,14 +2,16 @@ import { Route, Routes } from "react-router-dom";
...
-import { Home } from "./Pages";
+import { Home, Courses, Error } from "./Pages";
 
 const App = () => {
   return (
     <>
       <Navbar />
       <Routes>
-        <Route path="/" element={<Home />} />
+        <Route path="*" element={<Error />} />
```
Comment (codadept) [inline src/App.jsx:12]: Error component should be the last route. Move it to the bottom.

**Review by codadept — CHANGES_REQUESTED** [review]
Do the change and then the PR is ready to merge

File: src/App.jsx (hunk 6 above)
Comment (XRegiGigaSX) [reply src/App.jsx:12]: Fixed

**Review by codadept — APPROVED** [review]

**Pull request merged**